contracts/common/Multicall.sol

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Mostly taken from Uniswap/v3-periphery repository, with payable removed and solidity version
+// adjustments
+pragma solidity 0.7.3;
+pragma experimental "ABIEncoderV2";
+import "./MetaTransactionMsgSender.sol";
+
+/// @title Multicall
+/// @notice Enables calling multiple methods in a single call to the contract
+abstract contract Multicall is MetaTransactionMsgSender {
+
+
+    function multicall(bytes[] calldata data) public returns (bytes[] memory results) {
+        // First off, is this a metatransaction?
+        address sender = msgSender();
+        bytes memory affix;
```

Comment:
Would prefer this as a ternary, but that's just a personal style preference:

```
bytes memory affix = (msg.sender == address(this) && msgSender() != address(this)
    ? abi.encodePacked(METATRANSACTION_FLAG, msgSender())
    : "0x0"
```

Reply:
I know you like ternaries more than I do, but even so... do you really think yours is easier to reason about/understand than mine?

Reply:
I think it's a bit more succinct, but as I said, it's a style choice I leave to you

Reply:
If we were code golfing, I'd be all for it. But we're trying to write an easily-maintainable (and therefore easily understandable) codebase.